packages/aa/src/index.js

```diff
@@ -52,25 +52,53 @@ async function loadCanonicalNameMap ({ rootDir, includeDevDeps, resolve } = {})
   return canonicalNameMap
 }
 
+const depPackageJsonPathCache = new Map();
+function memoResolveSync (resolve, depName, packageDir) {
+  const key = depName + '!' + packageDir;
+  if (depPackageJsonPathCache.has(key)) {
+    return depPackageJsonPathCache.get(key)
+  } else {
+    const depRelativePackageJsonPath = path.join(depName, 'package.json')
+    let depPackageJsonPath
+    // If this function used async, it'd have to be awaited, which would mean cache lookup 
+    // would need to happen outside the function to save on performance of spawning a promise 
+    // for each cache lookup.
+    depPackageJsonPath = resolve.sync(depRelativePackageJsonPath, { basedir: packageDir })
+    depPackageJsonPathCache.set(key, depPackageJsonPath)
+    return depPackageJsonPath
+  }
+}
+const depsToWalkCache = new Map();
+function memoListDependencies (packageDir, includeDevDeps) {
+  const key = packageDir + (includeDevDeps ? '-D' : '')
+  if (depsToWalkCache.has(key)) {
+    return depsToWalkCache.get(key)
+  } else {
+    const packageJsonPath = path.join(packageDir, 'package.json')
+    // If this function used async, it'd have to be awaited, which would mean cache lookup 
+    // would need to happen outside the function to save on performance of spawning a promise 
+    // for each cache lookup.
+    const rawPackageJson = readFileSync(packageJsonPath, 'utf8')
+    const packageJson = JSON.parse(rawPackageJson)
+    const depsToWalk = [
+      ...Object.keys(packageJson.dependencies || {}),
```

Comment:
`.dependencies` makes the function that reads it megamorphic, so there's theoretically a performance benefit from it, but I didn't see much. 
The biggest performance issues were:
- resolve.sync making statSync checks (which throws and catches internally for each location where the file is not, so it's awfully slow)
- reading and parsing json many times around.